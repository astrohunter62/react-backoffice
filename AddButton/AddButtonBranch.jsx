import React from 'react'
import PropTypes from 'prop-types'

import { withStyles } from 'material-ui/styles'
import Button from 'material-ui/Button'
import AddIcon from 'material-ui-icons/Add'

const styles = theme => ({
  button: {
    position: 'fixed',
    right: theme.spacing.unit * 3,
    bottom: theme.spacing.unit * 3,
  },
})

const AddButton = ({ handleClick, classes }) => {
  return (
    <Button
      fab
      color="accent"
      aria-label="add"
      className={classes.button}
      onClick={handleClick}
    >
      <AddIcon />
    </Button>
  )
}

AddButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
  classes: PropTypes.object.isRequired,
}

export default withStyles(styles)(AddButton)
