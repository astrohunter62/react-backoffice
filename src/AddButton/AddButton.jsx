import React from 'react'
import PropTypes from 'prop-types'

import {
  withStyles,
  Fab,
} from '@material-ui/core'
import AddIcon from '@material-ui/icons/Add'

const styles = theme => ({
  button: {
    position: 'fixed',
    right: theme.spacing.unit * 3,
    bottom: theme.spacing.unit * 3,
  },
})

const AddButton = ({ onClick, classes }) => (
  <Fab
    color="secondary"
    aria-label="add"
    className={classes.button}
    onClick={onClick}
  >
    <AddIcon />
  </Fab>
)

AddButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

export default withStyles(styles)(AddButton)
