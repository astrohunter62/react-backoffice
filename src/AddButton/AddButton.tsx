import React from "react";
import { withStyles, Fab } from "@material-ui/core";
import AddIcon from "@material-ui/icons/Add";

const styles = (theme: any) => ({
  button: {
    position: "fixed",
    right: theme.spacing(3),
    bottom: theme.spacing(3),
  },
});

type AddButtonProps = {
  onClick: (...args: any[]) => any;
  classes: {
    [key: string]: string;
  };
};

const AddButton: React.SFC<AddButtonProps> = ({ onClick, classes }) => (
  <Fab
    color="secondary"
    aria-label="add"
    className={classes.button}
    onClick={onClick}
  >
    <AddIcon />
  </Fab>
);
export default withStyles(styles as any)(AddButton);
