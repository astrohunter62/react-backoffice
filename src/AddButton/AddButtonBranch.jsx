import React from 'react'
import PropTypes from 'prop-types'

import {
  Button,
  AddIcon,
  withStyles,
} from '@material-ui/core'

const styles = theme => ({
  button: {
    position: 'fixed',
    right: theme.spacing.unit * 3,
    bottom: theme.spacing.unit * 3,
  },
})

const AddButton = ({ onClick, classes }) => (
  <Button
    variant="fab"
    color="secondary"
    aria-label="add"
    className={classes.button}
    onClick={onClick}
  >
    <AddIcon />
  </Button>
)

AddButton.propTypes = {
  onClick: PropTypes.func.isRequired,
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
}

export default withStyles(styles)(AddButton)
