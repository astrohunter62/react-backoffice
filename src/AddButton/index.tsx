import React, { FunctionComponent } from "react";
import { Fab, Theme, makeStyles } from "@material-ui/core";
import AddIcon from "@material-ui/icons/Add";

const useStyles = makeStyles((theme: Theme) => ({
  button: {
    position: "fixed",
    right: theme.spacing(3),
    bottom: theme.spacing(3),
  },
}));

type AddButtonProps = {
  onClick: (...args: any[]) => any;
};

const AddButton: FunctionComponent<AddButtonProps> = ({ onClick }) => {
  const classes = useStyles();

  return (
    <Fab
      color="secondary"
      aria-label="add"
      className={classes.button}
      onClick={onClick}
    >
      <AddIcon />
    </Fab>
  );
};

export default AddButton;
