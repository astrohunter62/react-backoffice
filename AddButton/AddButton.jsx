import React from 'react'
import PropTypes from 'prop-types'

import AddButtonBranch from './AddButtonBranch'

const withAddButton = (Component) => class extends React.Component {
  render() {
    return (
      <Component {...this.props} />
    )
  }
}

const AddButton = withAddButton(AddButtonBranch)

AddButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
}

export default AddButton
