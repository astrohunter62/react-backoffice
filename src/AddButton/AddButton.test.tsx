import React from "react";
import { shallow } from "enzyme";

import AddButton from ".";

it("renders correctly", () => {
  const tree = shallow(<AddButton onClick={() => {}} />);

  expect(tree).toMatchSnapshot();
});
