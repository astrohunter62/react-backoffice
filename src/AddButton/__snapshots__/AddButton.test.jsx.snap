// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  aria-label="add"
  className="MuiButtonBase-root-24 MuiButton-root-2 MuiButton-contained-11 MuiButton-fab-19 MuiButton-containedSecondary-13 MuiButton-raised-14 MuiButton-raisedSecondary-16 AddButton-button-1"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-3"
  >
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root-27"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <g>
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </g>
    </svg>
  </span>
  <span
    className="MuiTouchRipple-root-33"
  />
</button>
`;
